### ISSUE TITLE
Pseudo Label in Distill Whisper

### ISSUE THREAD

**Issue by nhduong1203** [innerNULL/simpler-distil-whisper#1]
Hello, thanks for your simplified version of distill-whisper. It's very easy to follow. However, I believe one of the key techniques in distill-whisper is using the pseudo labels generated by the original Whisper model as the labels for training the Pseudo Label loss. But it seems like you're using the true labels for training the Pseudo Label loss instead.

I understand that using true labels as hard labels is also a valid approach. However, since you've named it pl_loss, I think it would be more consistent if pseudo labels were used as the hard labels.

**Comment by innerNULL** [comment]
> Hello, thanks for your simplified version of distill-whisper. It's very easy to follow. However, I believe one of the key techniques in distill-whisper is using the pseudo labels generated by the original Whisper model as the labels for training the Pseudo Label loss. But it seems like you're using the true labels for training the Pseudo Label loss instead.
> 
> I understand that using true labels as hard labels is also a valid approach. However, since you've named it pl_loss, I think it would be more consistent if pseudo labels were used as the hard labels.

Hi~ haha it's long time ago as I've even forgot this work : D

I think here's the place loading pseudo label [code](https://github.com/innerNULL/simpler-distil-whisper/blob/1fb658daa9ac2cac7d685091ab9c5efa21bd36b2/run_distillation.py#L319C22-L319C34), you can see column name of pseudo label can be defined in configs file, the default value is `"text_col": "text"`, and when you run `run_pseudo_labelling.py` with default configs, you can find the pseudo label will be saved in "text" field of output JSON lines file (see `"target_text_col": "text"` in `run_pseudo_labelling.json`). So I guess should be correct haha?

But yes, I think the pseudo label names of configs are a little confusing🤣

**Comment by nhduong1203** [comment]
I think we should use the pseudo label for PL loss and true label for KD loss. But you use the same label for both PL loss and KD loss.

**Comment by innerNULL** [comment]
> I think that we should use the pseudo label for PL loss and true label for KD loss. But you use the same label for both 2 PL loss and KD loss.

ah got your point, let me check

**Comment by innerNULL** [comment]
> I think we should use the pseudo label for PL loss and true label for KD loss. But you use the same label for both PL loss and KD loss.

Actully I found in official implementation they also using pseudo label to calculate both logits and loss, can refer [here](https://github.com/huggingface/distil-whisper/blob/9422916d714bde70ea9d88e13c56f877ced1e76b/training/run_distillation.py#L1472)
<img width="917" alt="image" src="https://github.com/user-attachments/assets/8dc767e3-b028-4765-9556-bc522c53e5bc">

**Comment by nhduong1203** [comment]
I see. Actually we don't need any label to compute KD Loss. This mean we don't need to pass label into this or any label is valid.
```
student_model_outputs: Duct[str, Tensor] = student_model(
    input_features=inputs, labels=labels
)
```
Thank you.

**Comment by innerNULL** [comment]
> I see. Actually we don't need any label to compute KD Loss. This mean we don't need to pass label into this or any label is valid.
> 
> ```
> student_model_outputs: Duct[str, Tensor] = student_model(
>     input_features=inputs, labels=labels
> )
> ```
> 
> Thank you.

exactly, the only thing we need is logits, here we pass `labels` cuz we can get everything we need in one pass~
